src/tests/netsim_tests.rs

```diff
@@ -0,0 +1,389 @@
+
```

Comment:
We should probably add license header.

Reply:
Also, if file path is `src/tests/netsim_tests.rs`, do we have to say tests again? Sounds like redundant labeling for me, I'd go with `src/tests/netsim.rs` :)